pkg/destroy/gcp/network.go

```diff
@@ -89,27 +101,39 @@ func (o *ClusterUninstaller) deleteAddress(name string) error {
 	o.Logger.Debugf("Deleting address %s", name)
 	ctx, cancel := o.contextWithTimeout()
 	defer cancel()
-	_, err := o.computeSvc.Addresses.Delete(o.ProjectID, o.Region, name).Context(ctx).Do()
+	resp, err := o.computeSvc.Addresses.Delete(o.ProjectID, o.Region, name).RequestId(o.requestID("address", name)).Context(ctx).Do()
 	if err != nil && !isNoOp(err) {
+		o.resetRequestID("address", name)
 		return errors.Wrapf(err, "failed to delete address %s", name)
 	}
+	if resp.Status == "DONE" && isErrorStatus(resp.HttpErrorStatusCode) {
+		o.resetRequestID("address", name)
+		return errors.Errorf("failed to delete address %s with status code: %d", name, resp.HttpErrorStatusCode)
+	}
 	return nil
```

Comment:
based on 
>Given that resources do not immediately go away, we cannot for sure say that an item has been deleted unless one of 2 things happen:
    The long-running operation returned from the delete call changes its status to "DONE" and its HttpErrorStatusCode is not an error code.

wouldn't it be useful to return an error until the resource was actually deleted.. like pending deletion..

that should remove the need for `setFoundItems` ???

Reply:
Returning an error that indicates pending deletion is fine. However, that doesn't get rid of the need for keeping track of which items are pending/have been deleted.

Reply:
> However, that doesn't get rid of the need for keeping track of which items are pending/have been deleted.

the we can report the item was delete when the error is nil from `delete<type>`?? WHy would track still?

Reply:
Because I don't know whether something was deleted unless I know it existed before. After attempting to delete 5 instances, I return with 5 pending, then try again, and now I get no instances, return nil... how do I know that I deleted anything?

Reply:
Ahh, I see we want to capture the resources whose delete was started, and it got deleted before we had chance to check back and next List they don't exist.

+1